src/mito2/src/sst/parquet/reader.rs

```diff
@@ -256,13 +245,73 @@ impl ParquetReaderBuilder {
 
         Ok(metadata)
     }
+
+    /// Computes row groups to read.
+    async fn row_groups_to_read(
+        &self,
+        read_format: &ReadFormat,
+        parquet_meta: &ParquetMetaData,
+        metrics: &mut Metrics,
+    ) -> BTreeSet<usize> {
+        let mut row_group_ids: BTreeSet<_> = (0..parquet_meta.num_row_groups()).collect();
+        metrics.num_row_groups_unfiltered += row_group_ids.len();
+
+        // Applies index to prune row groups.
+        //
+        // TODO(zhongzc): Devise a mechanism to enforce the non-use of indices
+        // as an escape route in case of index issues, and it can be used to test
+        // the correctness of the index.
+        if let Some(index_applier) = &self.index_applier {
```

Comment:
Looks interesting!  I wonder if the order of these two pruning processes may impact the pruning performance.

Reply:
The overhead of the inverted index is fixed unless skipped.
If the overhead of the min-max index is affected by the count of pending row groups, then deferred min-max indexing would be better.

You reminded me that I should use row groups filtered by the inverted index for pruning with the min-max index.